datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -408,26 +392,6 @@ fn optimize_projections(
     }
 }
 
-/// This function applies the given function `f` to the projection indices
```

Comment:
All of those code was moved into `RequiredIndices` as methods both to encapsulate the logic a bit more as well as to allow creation without copying